fv3gfs-physics/fv3gfs/physics/_config.py

```diff
@@ -0,0 +1,125 @@
+    rad_snow: bool = (
+        NamelistDefaults.rad_snow
+    )  # consider snow in cloud fraction calculation
+    rad_rain: bool = (
+        NamelistDefaults.rad_rain
+    )  # consider rain in cloud fraction calculation
+    dw_ocean: float = NamelistDefaults.dw_ocean  # base value for ocean
+    dw_land: float = (
+        NamelistDefaults.dw_land
+    )  # base value for subgrid deviation / variability over land
+    # cloud scheme 0 - ?
+    # 1: old fvgfs gfdl) mp implementation
+    # 2: binary cloud scheme (0 / 1)
+    tau_l2v: float = (
+        NamelistDefaults.tau_l2v
+    )  # cloud water to water vapor (evaporation)
+    c2l_ord: int = NamelistDefaults.c2l_ord
+    do_sedi_heat: bool = NamelistDefaults.do_sedi_heat
+    do_sedi_w: bool = NamelistDefaults.do_sedi_w
+    fast_sat_adj: bool = NamelistDefaults.fast_sat_adj
+    qc_crt: float = NamelistDefaults.qc_crt
+    fix_negative: bool = NamelistDefaults.fix_negative
+    irain_f: int = NamelistDefaults.irain_f
+    mp_time: float = NamelistDefaults.mp_time
+    prog_ccn: bool = NamelistDefaults.prog_ccn
+    qi0_crt: float = NamelistDefaults.qi0_crt
+    qs0_crt: float = NamelistDefaults.qs0_crt
+    rh_inc: float = NamelistDefaults.rh_inc
+    rh_inr: float = NamelistDefaults.rh_inr
+    # rh_ins: Any
+    rthresh: float = NamelistDefaults.rthresh
+    sedi_transport: bool = NamelistDefaults.sedi_transport
+    # use_ccn: Any
+    use_ppm: bool = NamelistDefaults.use_ppm
+    vg_max: float = NamelistDefaults.vg_max
+    vi_max: float = NamelistDefaults.vi_max
+    vr_max: float = NamelistDefaults.vr_max
+    vs_max: float = NamelistDefaults.vs_max
+    z_slope_ice: bool = NamelistDefaults.z_slope_ice
+    z_slope_liq: bool = NamelistDefaults.z_slope_liq
+    tice: float = NamelistDefaults.tice
+    alin: float = NamelistDefaults.alin
+    clin: float = NamelistDefaults.clin
+
+    @classmethod
+    def from_f90nml(cls, namelist: f90nml.Namelist):
+        namelist_dict = namelist_to_flatish_dict(namelist.items())
+        namelist_dict = {
+            key: value
+            for key, value in namelist_dict.items()
+            if key in cls.__dataclass_fields__  # type: ignore
+        }
+        return cls(**namelist_dict)
+
+
+def namelist_to_flatish_dict(nml_input):
+    nml = dict(nml_input)
+    for name, value in nml.items():
+        if isinstance(value, f90nml.Namelist):
+            nml[name] = namelist_to_flatish_dict(value)
+    flatter_namelist = {}
+    for key, value in nml.items():
+        if isinstance(value, dict):
+            for subkey, subvalue in value.items():
+                if subkey in flatter_namelist:
+                    raise ValueError(
+                        "Cannot flatten this namelist, duplicate keys: " + subkey
+                    )
+                flatter_namelist[subkey] = subvalue
+        else:
+            flatter_namelist[key] = value
+    return flatter_namelist
```

Comment:
`namelist_to_flatish_dict` should probably go to `util` since it's used identically in the dycore. This can also live in `pace.util.namelist`.

Reply:
Moved to `pace.util.namelist`